packages/passport-interface/src/PassportPopup.ts

```diff
@@ -59,13 +59,19 @@ export function usePassportPopupSetup() {
       // Later, the Passport redirects back with a proof. Send it to our parent.
       window.opener.postMessage({ encodedPCD: paramsProof }, "*");
       window.close();
+      setTimeout(() => {
+        setError("Finished. Please close this window.");
+      }, 1000 * 3);
     } else if (paramsEncodingPendingPCD != null) {
       // Later, the Passport redirects back with a encodedPendingPCD. Send it to our parent.
       window.opener.postMessage(
         { encodedPendingPCD: paramsEncodingPendingPCD },
         "*"
       );
       window.close();
+      setTimeout(() => {
+        setError("Finished. Please close this window.");
+      }, 1000 * 3);
```

Comment:
minor but i'd pull this duplicated block into a local

```
function returnProofAndExit(proof) { ... }
```

so that here u just call

```
returnProofAndExit({ encodedPendingPCD: paramsEncodingPendingPCD });
```

Reply:
I'm going to leave it the way it is, I actually think it's more readable although a bit repetitive.